src/trees/SerialFunctionTree.cpp

```diff
@@ -562,72 +569,65 @@ template <int D> void SerialFunctionTree<D>::rewritePointers(int nChunks) {
-                    node->children[i] = this->nodeChunks[n_ichunk] + n_inode;
-                }
-                this->nodeStackStatus[node->serialIx] = 1; // occupied
-            }
-        }
-        this->lastNode = this->nodeChunks[ichunk] + this->nNodes % (this->maxNodesPerChunk);
-    }
-
-    // update other MWTree data
-    auto *Tree = static_cast<FunctionTree<D> *>(this->tree_p);
 
-    NodeBox<D> &rBox = Tree->getRootBox();
+    NodeBox<D> &rBox = this->getTree()->getRootBox();
     MWNode<D> **roots = rBox.getNodes();
 
+    int DepthMax = 100, slen = 0;
+    ProjectedNode<D> *stack[DepthMax * 8];
     for (int rIdx = 0; rIdx < rBox.size(); rIdx++) {
         roots[rIdx] = (this->nodeChunks[0]) + rIdx; // adress of roots are at start of NodeChunks[0] array
+        stack[slen++] = (this->nodeChunks[0]) + rIdx;
     }
+    this->nNodes = 0;
+    while (slen) {
+        ProjectedNode<D> *node = stack[--slen];
+        for (int i = 0; i < node->getNChildren(); i++) {
+            int n_ichunk = (node->childSerialIx + i) / this->maxNodesPerChunk;
+            int n_inode = (node->childSerialIx + i) % this->maxNodesPerChunk;
+            stack[slen++] = this->nodeChunks[n_ichunk] + n_inode;
+        }
+        int ichunk = node->serialIx / this->maxNodesPerChunk;
+        int inode = node->serialIx % this->maxNodesPerChunk;
 
-    this->getTree()->resetEndNodeTable();
-}
+        this->nNodes = std::max(this->nNodes, ichunk * this->maxNodesPerChunk + inode + 1);
+        this->getTree()->incrementNodeCount(node->getScale());
+        if (node->isEndNode()) this->getTree()->squareNorm += node->getSquareNorm();
+        if (node->isEndNode()) this->getTree()->endNodeTable.push_back(node);
 
-template <int D> int SerialFunctionTree<D>::getNChunksUsed() const {
-    int lastUsed = 0;
-    for (int iChunk = 0; iChunk < getNChunks(); iChunk++) {
-        int iShift = iChunk * this->maxNodesPerChunk;
-        bool chunkUsed = false;
-        for (int i = 0; i < this->maxNodesPerChunk; i++) {
-            if (this->nodeStackStatus[iShift + i] == 1) { chunkUsed = true; }
+        // normally (intel) the virtual table does not change, but we overwrite anyway
+        *(char **)(node) = this->cvptr_ProjectedNode;
+
+        node->tree = this->getTree();
+
+        //"adress" of coefs is the same as node, but in another array
+        node->coefs = this->nodeCoeffChunks[ichunk] + inode * this->sizeNodeCoeff;
+
+        // adress of parent and children must be corrected
+        // can be on a different chunks
+        if (node->parentSerialIx >= 0) {
+            int n_ichunk = node->parentSerialIx / this->maxNodesPerChunk;
+            int n_inode = node->parentSerialIx % this->maxNodesPerChunk;
+            node->parent = this->nodeChunks[n_ichunk] + n_inode;
+            assert(node->parent->serialIx == node->parentSerialIx);
+        } else {
+            node->parent = 0;
```

Comment:
`nullptr`

Reply:
ok, I corrected a bunch of them at the same time